sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2351,36 +2400,54 @@ def run(self):
         port_mapping_data, retry_eeprom_set = self.init()
 
         # Start the CMIS manager
-        cmis_manager = CmisManagerTask(self.namespaces, port_mapping_data, self.skip_cmis_mgr)
-        cmis_manager.task_run()
+        cmis_manager = CmisManagerTask(self.namespaces, port_mapping_data, self.stop_event, self.skip_cmis_mgr)
+        cmis_manager.start()
+        self.threads.append(cmis_manager)
 
         # Start the dom sensor info update thread
-        dom_info_update = DomInfoUpdateTask(self.namespaces, port_mapping_data)
-        dom_info_update.task_run()
+        dom_info_update = DomInfoUpdateTask(self.namespaces, port_mapping_data, self.stop_event)
+        dom_info_update.start()
+        self.threads.append(dom_info_update)
 
-        # Start the sfp state info update process
-        sfp_state_update = SfpStateUpdateTask(self.namespaces, port_mapping_data, retry_eeprom_set)
-        sfp_state_update.task_run(self.sfp_error_event)
-
-        # Start the Y-cable state info update process if Y cable presence established
+        # Start the sfp state info update thread
+        sfp_state_update = SfpStateUpdateTask(self.namespaces, port_mapping_data, retry_eeprom_set, self.stop_event, self.sfp_error_event)
+        sfp_state_update.start()
+        self.threads.append(sfp_state_update)
 
         # Start main loop
-        self.log_info("Start daemon main loop")
+        self.log_info("Start daemon main loop with thread count {}".format(len(self.threads)))
 
         self.stop_event.wait()
 
         self.log_info("Stop daemon main loop")
 
+        generate_sigkill = False
+        # check all threads are alive
+        for thread in self.threads:
```

Comment:
will the main thread continue monitor the child threads forever ?
if not 
suggest using stop_event.wait() to wait for the exception to be set.

Reply:
ok, I din see stop_event.wait() is put just before.